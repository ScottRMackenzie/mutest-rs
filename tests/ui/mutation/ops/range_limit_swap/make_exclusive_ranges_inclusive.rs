//@ print-mutations
//@ build
//@ stdout
//@ stderr: empty
//@ mutation-operators: range_limit_swap

fn f() {
    let arr = [0, 1, 2, 3, 4];

    for _ in 0..5 {}
    for _ in &arr[..3] {}

    match (&arr[..3], &arr[1..3]) {
        (_, _) => {}
    }

    if arr[..3].len() == arr[1..3].len() {}

    fn g<T>(_: &[T]) {}

    g(&arr[..3]);
    g(&arr[1..3]);
}

#[test]
fn test() {
    f();
}
