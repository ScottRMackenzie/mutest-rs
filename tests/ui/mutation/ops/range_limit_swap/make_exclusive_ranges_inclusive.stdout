1 mutation
  - [range_limit_swap] include limit in range expression in f at tests/ui/mutation/ops/range_limit_swap/make_exclusive_ranges_inclusive.rs:10:14: 10:18
    <-(0)- test

1 mutation
  - [range_limit_swap] include limit in range expression in f at tests/ui/mutation/ops/range_limit_swap/make_exclusive_ranges_inclusive.rs:11:19: 11:22
    <-(0)- test

1 mutation
  - [range_limit_swap] include limit in range expression in f at tests/ui/mutation/ops/range_limit_swap/make_exclusive_ranges_inclusive.rs:13:17: 13:20
    <-(0)- test

1 mutation
  - [range_limit_swap] include limit in range expression in f at tests/ui/mutation/ops/range_limit_swap/make_exclusive_ranges_inclusive.rs:13:28: 13:32
    <-(0)- test

1 mutation
  - [range_limit_swap] include limit in range expression in f at tests/ui/mutation/ops/range_limit_swap/make_exclusive_ranges_inclusive.rs:17:12: 17:15
    <-(0)- test

1 mutation
  - [range_limit_swap] include limit in range expression in f at tests/ui/mutation/ops/range_limit_swap/make_exclusive_ranges_inclusive.rs:17:30: 17:34
    <-(0)- test

1 mutation
  - [range_limit_swap] include limit in range expression in f at tests/ui/mutation/ops/range_limit_swap/make_exclusive_ranges_inclusive.rs:21:12: 21:15
    <-(0)- test

1 mutation
  - [range_limit_swap] include limit in range expression in f at tests/ui/mutation/ops/range_limit_swap/make_exclusive_ranges_inclusive.rs:22:12: 22:16
    <-(0)- test

8 mutants; 8 mutations; 8 safe; 0 unsafe (0 tainted); 0 batched; 8 unbatched
